### PR TITLE
doc/release-note: flash API & MCUboot notes

### PR DESCRIPTION
**Pull request by nvlsianpu**
Added record on MCUboot changes and flash API
changes which was inherited within the last upmerge
#4370.

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/releases/release-notes-latest.rst [hunk 1]
```diff
@@ -161,6 +167,17 @@ The following list summarizes the most important changes inherited from upstream
 
     * Implemented workaround for nRF52 anomaly 242 in the nRF SoC flash driver.
     * Added automatic selection of :option:`CONFIG_MPU_ALLOW_FLASH_WRITE` when the MPU is enabled for Arm based SoCs.
+    * :c:func:`flash_write_protection_set()` has been deprecated and will be removed in
```
Comment (nvlsianpu) [inline doc/nrf/releases/release-notes-latest.rst:170]: @ru-fu This is original zephyr record. It can be compressed.

**Review by ru-fu — CHANGES_REQUESTED** [review]
I'll add a commit.

File: doc/nrf/releases/release-notes-latest.rst [hunk 2]
```diff
@@ -161,6 +167,17 @@ The following list summarizes the most important changes inherited from upstream
```
Comment (ru-fu) [inline doc/nrf/releases/release-notes-latest.rst:167]: Does the commit need to be updated?

**Comment by ru-fu** [comment]
I added a commit - please check and squash, if okay.

**Comment by nvlsianpu** [comment]
@ru-fu done.

**Review by ru-fu — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**